workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowExecutionAspect.java

```diff
@@ -82,6 +82,10 @@ public WorkReport executeAroundAdvice(ProceedingJoinPoint proceedingJoinPoint, W
 
 		/* get workflow definition entity */
 		WorkFlowDefinition workFlowDefinition = this.workFlowDefinitionRepository.findFirstByName(workflowName);
+		if (workFlowDefinition == null) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("Cannot find workflow '"+ workflowName + "'"));
```

Comment:
Can you use WorkflowDefinitionException or create
WorkflowDefinitionNotFoundException ?